src/main/java/org/folio/rest/helper/FundsHelper.java

```diff
@@ -219,21 +231,26 @@ public Future<Void> updateFund(CompositeFund compositeFund, RequestContext reque
   private Future<Void> updateFundGroups(CompositeFund compositeFund, RequestContext requestContext) {
     Fund fund = compositeFund.getFund();
     Set<String> groupIds = new HashSet<>(compositeFund.getGroupIds());
+    log.debug("updateFundGroups:: Updating fund groups for compositeFund with id={}", fund.getId());
+
     return ledgerDetailsService.getCurrentFiscalYear(fund.getLedgerId(), new RequestContext(ctx, okapiHeaders))
       .compose(currentFiscalYear-> {
         if(Objects.nonNull(currentFiscalYear)) {
           String currentFiscalYearId = currentFiscalYear.getId();
+          log.info("updateFundGroups:: Retrieving group fund fiscal years for fundId={}, currentFiscalYearId={}", fund.getId(), currentFiscalYearId);
           return getGroupFundFiscalYearsThatFundBelongs(fund.getId(), currentFiscalYearId)
             .compose(groupFundFiscalYearCollection -> {
               List<String> groupIdsFromStorage = StreamEx.of(groupFundFiscalYearCollection).map(GroupFundFiscalYear::getGroupId).collect(Collectors.toList());
-
+              log.info("updateFundGroups:: Creating group fund fiscal years for new groups");
               return createGroupFundFiscalYears(compositeFund, currentFiscalYearId, getSetDifference(groupIdsFromStorage, groupIds), requestContext)
                 .compose(vVoid -> deleteGroupFundFiscalYears(groupFundFiscalYearIdsForDeletion(
                   groupFundFiscalYearCollection, getSetDifference(groupIds, groupIdsFromStorage))));
             });
         } else if(groupIds.isEmpty()) {
+          log.warn("updateFundGroups:: GroupIds is empty");
```

Comment:
even we have fundId in the info level that can be matched to all below logs in this method, better to add fundId to this log and log below, because for example if we know that something is wrong with particular fundId - we can use it as a keyphrase to find these 2 error logs easily without requesting full log for the specified period